libbeat/autodiscover/providers/docker/docker.go

```diff
@@ -383,7 +383,7 @@ func (d *Provider) generateHints(event bus.Event) bus.Event {
 		e["ports"] = ports
 	}
 	if labels, err := dockerMeta.GetValue("labels"); err == nil {
-		hints := utils.GenerateHints(labels.(mapstr.M), "", d.config.Prefix)
+		hints, _ := utils.GenerateHints(labels.(mapstr.M), "", d.config.Prefix, false, []string{})
```

Comment:
Why don't we log incorrect hints here? I assume because of the `false` flag was passed. Should we add a comment explaining this?

I'd rather use an option pattern here like `WithHintsValidation(AllSupportedHints)` instead of passing a boolean value but I can see that this interface is coming from https://github.com/elastic/elastic-agent-autodiscover/pull/89 and cannot be changed in this PR.

Reply:
I have updated also the docker to support validation
I have added a comment in nomad.go here https://github.com/elastic/beats/pull/38213/files#diff-5e91552d1e0877c2c34d815ed28cd89530f485d16f6ddd362785bdf610f05fecR280 where we keep the validation disabled

Mainly the driver of this [story](https://github.com/elastic/elastic-agent/issues/3064) was to introduce the feature in elastic-agent and especially in kubernetes